### ISSUE TITLE
[Bug]: overflow: scroll doesn't apply to headers until actually scrolled

### ISSUE THREAD

**Issue by Timppe** [handsontable/handsontable#10469]
### Describe the bug

When explictly setting the overflow: scroll to the table `.ht_master .wtHolder {
    overflow: scroll;
  }` but not having anything to scroll yet, the header is shown on top of the scrollbar and the whole table is e.g. shifted when scrolling to the end.

### Video/Screenshots

When opening the window:
![image](https://github.com/handsontable/handsontable/assets/13637338/f238868f-fc38-420b-8ee2-1ed3ce431c95)

When scrolled to the bottom:
![image](https://github.com/handsontable/handsontable/assets/13637338/14feeafb-0c62-4de9-b014-9c563a04650e)

Not yet fixed after simply resizing the window to lure out scrolling need:
![image](https://github.com/handsontable/handsontable/assets/13637338/6b3c8172-7bc4-45cb-b442-743b81a67531)

Only after scrolling horizontally a bit, the header is adjusted correctly:
![image](https://github.com/handsontable/handsontable/assets/13637338/b45e93b9-0f4a-4d48-9d0c-4eca5bec08bd)

### Provide a link to the demo with the bug reproduction

https://jsfiddle.net/08oen5q9/

### Handsontable version

13.0. (latest)

### Your environment

Windows, Chrome, Edge, Firefox

**Comment by adrianszymanski89** [comment]
Hi @Timppe 

I am not able to reproduce this issue on my side. However, I see that you modified the styling of the core Handsontable elements, which is not recommended.

It should be done by adding an `overflow` parameter to the main container of the table - in this case, `example1`. Unfortunately, it doesn't work correctly at the moment. We have this issue reported here: https://github.com/handsontable/handsontable/issues/8321.

**Comment by AMBudnik** [comment]
I also wasn't able to replicate any artifacts. I've tried Win11 and macOS with Chrome 116.

**Comment by Timppe** [comment]
@adrianszymanski89 Yes, this bug is blocking us from doing a significant usability improvement to our application.

I don't understand how you can't see what I see. I just open the initial fiddle link and see this:
![image](https://github.com/handsontable/handsontable/assets/13637338/635d6cf3-5b56-49d6-9b3a-e03f759017cb)
And when scrolling the table, the last data row is clearly out of sync from the row header:
![image](https://github.com/handsontable/handsontable/assets/13637338/cca17f6f-84f8-45ad-95bd-4cab2798c435)
Same for my colleagues.

**Comment by adrianszymanski89** [comment]
Hi @Timppe 

I was finally able to get the same results as you. However, as I mentioned before, you are modifying core Handsontable styling, which is highly not recommended by us, and from what I can see, that is the cause of the issue here.  Until we fix https://github.com/handsontable/handsontable/issues/8321, I can't think of any workarounds regarding this.

**Comment by adrianspdev** [comment]
Hi @Timppe, thanks for the report.

I rebuilt your jsfiddle setup verbatim (flex container with min-width:200px; width:auto; flex-grow:1, .headered-table .ht_master .wtHolder { overflow-x: scroll }, no width passed to Handsontable, height: 320, 20×20 data) and ran it side-by-side against:

- v13.0.0 (the version you originally reported)
- v17.0.1 (current develop)

To rule out macOS overlay scrollbars hiding the issue I forced a 14px webkit scrollbar via ::-webkit-scrollbar, mimicking Windows native scrollbars.

Pixel-level measurements at the rightmost rendered column on both versions, on initial render, after window resize, and after shrinking the container:

released v13.0.0  headerRight=1066  bodyRight=1066  diff= 0px  ALIGNED
local    v17.0.1  headerRight=1066  bodyRight=1066  diff= 0px  ALIGNED
released shrunk   headerRight= 766  bodyRight= 766  diff= 0px  ALIGNED
local    shrunk   headerRight= 675  bodyRight= 675  diff= 0px  ALIGNED

Header always lines up with body, scrollbar gutter sits outside the table content, no shift after first horizontal scroll. I cannot reproduce the original behavior in either version.

Closing as no longer reproducible - feel free to reopen with an updated minimal repro on v17.0.1 if you still hit this.

Thanks again for filing this!
